client/fvESg_service.go

```diff
@@ -0,0 +1,389 @@
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
+			}
+		}
+	}`, "fvRsSecInherited", dn))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationfvRsSecInheritedFromEndpointSecurityGroup(parentDn , tDn string) error{
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	return sm.DeleteByDn(dn , "fvRsSecInherited")
+}
+
+func (sm *ServiceManager) ReadRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsSecInherited")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsSecInherited")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tDn")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsProvFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rsprov-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
+			}
+		}
+	}`, "fvRsProv", dn))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationfvRsProvFromEndpointSecurityGroup(parentDn , tnVzBrCPName string) error{
+	dn := fmt.Sprintf("%s/rsprov-%s", parentDn, tnVzBrCPName)
+	return sm.DeleteByDn(dn , "fvRsProv")
+}
+
+func (sm *ServiceManager) ReadRelationfvRsProvFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsProv")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsProv")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tnVzBrCPName")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsConsIfFromEndpointSecurityGroup( parentDn, tnVzCPIfName string) error {
+	dn := fmt.Sprintf("%s/rsconsIf-%s", parentDn, tnVzCPIfName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.